docs/source/advanced/sagemaker-pipelines.rst

```diff
@@ -0,0 +1,303 @@
+
+This command sets up a pipeline with three main stages:
+
+1. Graph construction using the configuration in ``my_gconstruct_config.json``
+2. Model training using the settings in ``train.yaml``
+3. Inference using the trained model
+
+The pipeline will use 2 instances (specified by ``--instance-count``) for distributed training and inference.
+
+You'll need to provide:
+
+* A SageMaker execution role (``--role``) with appropriate permissions
+* A GraphStorm Docker image (``--graphstorm-pytorch-cpu-image-url``) for running the tasks
+* S3 locations for your input data and where to store results
+
+Running Pipeline Executions
+-------------------------
+
+Once you've created a pipeline, you can execute it using the ``execute_sm_pipeline.py`` script:
+
+.. code-block:: bash
+
+    python execute_sm_pipeline.py \
+        --pipeline-name my-graphstorm-pipeline \
+        --region us-west-2
+
+You can override default parameters during execution to customize the run:
+
+.. code-block:: bash
+
+    python execute_sm_pipeline.py \
+        --pipeline-name my-graphstorm-pipeline \
+        --region us-west-2 \
+        --instance-count 4 \
+        --gpu-instance-type ml.g4dn.12xlarge
+
+Pipeline Components
+-----------------
+
+A GraphStorm SageMaker pipeline can include several components that you can combine based on your needs.
+We list those here, with the step name that you can provide in ``--jobs-to-run`` in parentheses.
+
+1. **Single-instance Graph Construction** (``gconstruct``):
+   Single-instance graph construction for smaller graphs.
+
+2. **Distributed Graph pre-processing** (``gsprocessing``):
+   PySpark-based distributed data preparation for large graphs.
+
+3. **Distributed Graph Partitioning** (``dist_part``):
+   Multi-instance graph partitioning for distributed training.
+
+4. **GraphBolt Conversion** (``gb_convert``):
+   Converts partitioned data to GraphBolt format for improved performance.
+
+5. **Training** (``train``):
+   Trains your graph neural network model.
+
+6. **Inference** (``inference``):
+   Runs predictions using your trained model.
+
+Configuration Options
+---------------------
+
+This section provides a comprehensive list of all available configuration options for creating and executing GraphStorm SageMaker pipelines.
+
+AWS Configuration
+^^^^^^^^^^^^^^^^^
+
+* ``--execution-role``: SageMaker execution IAM role ARN. (Required)
+* ``--region``: AWS region. (Required)
+* ``--graphstorm-pytorch-cpu-image-uri``: GraphStorm GConstruct/dist_part/train/inference CPU ECR image URI. (Required)
+* ``--graphstorm-pytorch-gpu-image-uri``: GraphStorm GConstruct/dist_part/train/inference GPU ECR image URI.
+* ``--gsprocessing-pyspark-image-uri``: GSProcessing SageMaker PySpark ECR image URI. (Required if running a ``gsprocessing`` job.)
+
+Instance Configuration
+^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--instance-count`` / ``--num-parts``: Number of worker instances/partitions for partition, training, inference. (Required)
```

Comment:
We should emphasize `--instance-count` and `--num-parts` must be same.

Reply:
There's only one argument, we're just using an alias to use both options (this makes it consistent with our sagemaker launch scripts and other documentation).

So the user can only set this once.